### ISSUE TITLE
"Deny all egress" policy does not restrict access to private address

### ISSUE THREAD

**Issue by libnux** [cloudnativelabs/kube-router#609]
**[K8s]**
version: v1.13
created with kubeadm with one master and one node

**[kube-router]**
version: v0.2.3
router,firewall and service-proxy are all enabled

**[How to produce]**
```
kubectl create ns netpol-test

cat <<EOF | kubectl apply -f -
apiVersion: extensions/v1beta1
kind: NetworkPolicy
metadata:
  name: default-deny
  namespace: netpol-test
spec:
  podSelector: {}
  policyTypes:
  - Egress
EOF

kubectl run -n netpol-test busybox --rm -ti --image=busybox /bin/sh
```

**[Current Result]**
In the busybox container, the CoreDNS and kubernetes apiserver endpoint were still accessible:
10.96.0.10 is CoreDNS address. 10.96.0.1 is the kubernetes apiserver.
```/ # ping 10.96.0.10
PING 10.96.0.10 (10.96.0.10): 56 data bytes
64 bytes from 10.96.0.10: seq=0 ttl=64 time=0.304 ms
64 bytes from 10.96.0.10: seq=1 ttl=64 time=0.141 ms

/ # wget -O - https://10.96.0.1
Connecting to 10.96.0.1 (10.96.0.1:443)
wget: note: TLS certificate validation not implemented
wget: server returned error: HTTP/1.1 403 Forbidden
```
**[Expected Result]**
All egress access should be restricted, including the service cidr, pod cidr and all other private addresses(10.0/8, 172.16/12,192.168/16)

**Comment by murali-reddy** [comment]
@libnux Why are you using old network policy API: `extensions/v1beta1` instead of `networking.k8s.io/v1`? Please try GA version of network policy and see.

**Comment by libnux** [comment]
@murali-reddy  Still got the same result:

```
libnux@server:~$ kubectl delete networkpolicy -n netpol-test default-deny
libnux@server:~$ cat <<EOF | kubectl apply -f -
> apiVersion: networking.k8s.io/v1
> kind: NetworkPolicy
> metadata:
>   name: default-deny
>   namespace: netpol-test
> spec:
>   podSelector: {}
>   policyTypes:
>   - Egress
> EOF
networkpolicy.networking.k8s.io/default-deny created
libnux@server:~$ kubectl run -n netpol-test busybox --rm -ti --image=busybox /bin/sh
kubectl run --generator=deployment/apps.v1 is DEPRECATED and will be removed in a future version. Use kubectl run --generator=run-pod/v1 or kubectl create instead.
        If you don't see a command prompt, try pressing enter.
/ # ping 10.96.0.10
PING 10.96.0.10 (10.96.0.10): 56 data bytes
64 bytes from 10.96.0.10: seq=0 ttl=64 time=0.271 ms
64 bytes from 10.96.0.10: seq=1 ttl=64 time=0.149 ms
^C
--- 10.96.0.10 ping statistics ---
2 packets transmitted, 2 packets received, 0% packet loss
round-trip min/avg/max = 0.149/0.210/0.271 ms
/ # wget -O - https://10.96.0.1
Connecting to 10.96.0.1 (10.96.0.1:443)
wget: note: TLS certificate validation not implemented
wget: server returned error: HTTP/1.1 403 Forbidden
/ #
```

**Comment by murali-reddy** [comment]
ok. i am able to observe same behaviour. Marking as bug.

**Comment by ramnes** [comment]
Seems identical than #474.

**Comment by ticpu** [comment]
This is fixed with nftables (pull request #745) if you want to give it a try.
we found that out later that services were restricted after switching firewall framework.
